core/primitives-core/src/profile.rs

```diff
@@ -85,7 +131,7 @@ impl ProfileData {
     }
 
     pub fn set_burnt_gas(&self, burnt_gas: u64) {
-        *self.data.borrow_mut().get_mut(0 as usize).unwrap() = burnt_gas;
+        self.with_data(|data| data[0] = burnt_gas);
```

Comment:
```suggestion
        self.add_val(0, burnt_gas);
```

Reply:
I'd rather keep the manual `=` -- otherwise, setting gas twice in a row would be surprising. Granted, one probably shouldn't set the gas twice right now, but who knows what happens with code in the future.

Reply:
Fair enough.